lib/src/map/options/enabled_gestures.dart

```diff
@@ -0,0 +1,254 @@
+  final bool doubleTapDragZoom;
+
+  /// Enable zooming with a mouse scroll wheel
+  final bool scrollWheelZoom;
+
+  /// Enable rotation by pressing the CTRL key and dragging with the cursor
+  /// or finger.
+  final bool ctrlDragRotate;
+
+  /// Returns true of any gesture with more than one finger is enabled.
+  bool hasMultiFinger() => twoFingerMove || twoFingerZoom || twoFingerRotate;
+
+  /// Wither to change the value of some gestures. Returns a new
+  /// [EnabledGestures] object.
+  EnabledGestures withFlag({
+    bool? drag,
+    bool? flingAnimation,
+    bool? twoFingerZoom,
+    bool? twoFingerMove,
+    bool? doubleTapZoomIn,
+    bool? doubleTapDragZoom,
+    bool? scrollWheelZoom,
+    bool? twoFingerRotate,
+    bool? ctrlDragRotate,
+  }) =>
+      EnabledGestures._(
+        drag: drag ?? this.drag,
+        flingAnimation: flingAnimation ?? this.flingAnimation,
+        twoFingerZoom: twoFingerZoom ?? this.twoFingerZoom,
+        twoFingerMove: twoFingerMove ?? this.twoFingerMove,
+        doubleTapZoomIn: doubleTapZoomIn ?? this.doubleTapZoomIn,
+        doubleTapDragZoom: doubleTapDragZoom ?? this.doubleTapDragZoom,
+        scrollWheelZoom: scrollWheelZoom ?? this.scrollWheelZoom,
+        twoFingerRotate: twoFingerRotate ?? this.twoFingerRotate,
+        ctrlDragRotate: ctrlDragRotate ?? this.ctrlDragRotate,
+      );
+
+  @override
+  bool operator ==(Object other) =>
+      identical(this, other) ||
+      other is EnabledGestures &&
+          runtimeType == other.runtimeType &&
+          drag == other.drag &&
+          flingAnimation == other.flingAnimation &&
+          twoFingerMove == other.twoFingerMove &&
+          twoFingerZoom == other.twoFingerZoom &&
+          doubleTapZoomIn == other.doubleTapZoomIn &&
+          doubleTapDragZoom == other.doubleTapDragZoom &&
+          scrollWheelZoom == other.scrollWheelZoom &&
+          twoFingerRotate == other.twoFingerRotate &&
+          ctrlDragRotate == other.ctrlDragRotate;
+
+  @override
+  int get hashCode => Object.hash(
+        drag,
+        flingAnimation,
+        twoFingerMove,
+        twoFingerZoom,
+        doubleTapZoomIn,
+        doubleTapDragZoom,
+        scrollWheelZoom,
+        twoFingerRotate,
+        ctrlDragRotate,
+      );
+}
+
+/// Use [InteractiveFlag] to disable / enable certain events Use
+/// [InteractiveFlag.all] to enable all events, use [InteractiveFlag.none] to
+/// disable all events
+///
+/// If you want mix interactions for example drag and rotate interactions then
+/// you have two options:
+///   a. Add your own flags:
+///      [InteractiveFlag.drag] | [InteractiveFlag.twoFingerRotate]
+///   b. Remove unnecessary flags from all:
+///     [InteractiveFlag.all] &
+///       ~[InteractiveFlag.flingAnimation] &
+///       ~[InteractiveFlag.twoFingerMove] &
+///       ~[InteractiveFlag.twoFingerZoom] &
+///       ~[InteractiveFlag.doubleTapZoomIn]
+abstract class InteractiveFlag {
```

Comment:
Am I missing something, or is having this in addition to `EnabledGestures` redundant?

Reply:
While writing the scope of this pull request I recignized that some really like the bitfield approach to define gestures. That's why I kept it in for everyone who prefers to define gestures that way. In case a user doesn't use this functionality it will get treeshakend.

Reply:
Do you want to have the bitfield operations removed or to keep them for those that prefer them?

Reply:
We should have only one system, otherwise it can be confusing.